### PR TITLE
fix(systemd): create WorkingDirectory via StateDirectory and harden register fallback

### LINKED ISSUE #325: fix(systemd): alpamon.service fails with 200/CHDIR when /var/lib/alpamon does not exist
## Summary

When alpamon is installed on a Linux + systemd host through the Cloud automated
install path (AWS SSM `AWS-RunShellScript`), the package and the systemd unit are
installed correctly but the service never starts. It enters a crash-restart loop
and ends in `failed`. The root cause is that `alpamon.service` declares
`WorkingDirectory=/var/lib/alpamon` but **nothing in the unit guarantees that
directory exists**, and on the Cloud install path the directory may never be
created before the service is started.

## Observed behavior

`systemctl status alpamon` / `journalctl -u alpamon` on the affected EC2 instance:

```
(alpamon)[101306]: alpamon.service: Changing to the requested working directory failed: No such file or directory
(alpamon)[101306]: alpamon.service: Failed at step CHDIR spawning /usr/bin/alpamon: No such file or directory
systemd[1]: alpamon.service: Main process exited, code=exited, status=200/CHDIR
systemd[1]: alpamon.service: Failed with result 'exit-code'.
systemd[1]: alpamon.service: Scheduled restart job, restart counter is at 5.
systemd[1]: alpamon.service: Start request repeated too quickly.
systemd[1]: Failed to start alpamon.service - Secure Server Agent for Alpacon.
```

- The package is installed: `/usr/bin/alpamon` exists.
- `status=200/CHDIR` is **not** a missing-binary error — it means systemd could not
  `chdir` into `WorkingDirectory` before `ExecStart`.
- `ls -ld /var/lib/alpamon` → directory does not exist.

## Expected behavior

After a Cloud automated install, `alpamon.service` should reach
`active (running)` and `/var/lib/alpamon` should exist with `root:root 0750`.

### 1. The unit does not create its own working directory

`configs/alpamon.service`:

```ini
[Service]
Type=simple
ExecStart=/usr/bin/alpamon
WorkingDirectory=/var/lib/alpamon   # assumed to pre-exist; nothing creates it here
```

There is no `StateDirectory=` (or equivalent) directive, so systemd assumes
`/var/lib/alpamon` already exists. Directory creation is delegated entirely to
out-of-unit steps:

- `scripts/postinstall.sh` → `alpamon setup` → `systemd-tmpfiles --create`, but only
  when `PLUGIN_ID`/`PLUGIN_KEY` env vars are set (`setup_alpamon()` returns non-zero
  and skips both setup and service start when they are unset).
- `alpamon register` → `ensureDirectories()` → `systemd-tmpfiles --create alpamon.conf`
  (`cmd/alpamon/command/register/service_linux.go`).

### 2. The Cloud install path can leave the directory uncreated

The Cloud scripts (`alpacon-server` `cloud/templates/cloud/install_script.sh` and
`cloud_init_script.sh`) use flag-based `alpamon register`, **without** the
`PLUGIN_ID`/`PLUGIN_KEY` env vars:

1. `apt-get install -y alpamon` runs postinstall. With no env vars,
   `setup_alpamon()` returns non-zero → `systemd-tmpfiles` is not run and the
   service is not started. **`/var/lib/alpamon` is not created here.**
2. `alpamon register …` is then expected to create the directory via
   `ensureDirectories()` and start the service.

So directory creation hinges solely on step 2's
`systemd-tmpfiles --create alpamon.conf`. If that does not actually create the
directory (e.g. the tmpfiles drop-in `/usr/lib/tmpfiles.d/alpamon.conf` is not
resolved/installed, and `systemd-tmpfiles` exits 0 without creating anything),
`register` proceeds to `systemctl start` → immediate `200/CHDIR` crash loop.

### 3. Failure becomes sticky on retry

- The install script's idempotency guard only checks `systemctl is-active`. A
  crash-looping unit is not active, so re-runs proceed.
- But the config file written on the first run already exists, so `register`
  aborts with "config file already exists" (`register.go`) **before** reaching
  `ensureDirectories()` — the directory is never created and the already-enabled
  failing unit keeps looping.

### A. (required) Add `StateDirectory` to `configs/alpamon.service`

```ini
[Service]
Type=simple
ExecStart=/usr/bin/alpamon
WorkingDirectory=/var/lib/alpamon
StateDirectory=alpamon
StateDirectoryMode=0750
SELinuxContext=-unconfined_u:unconfined_r:unconfined_t:s0
AppArmorProfile=-unconfined
Restart=on-failure
StandardOutput=journal
StandardError=journal
SyslogIdentifier=alpamon
```

- systemd creates `StateDirectory` **before** the `chdir` to `WorkingDirectory`, so
  the CHDIR failure disappears structurally.
- No `User=` is set, so the directory is created `root:root 0750`, matching the
  existing `configs/tmpfile.conf` and `postinstall.sh:create_directories()` — no
  ownership/permission change for existing installs.
- Without `DynamicUser`, `StateDirectory` is **persistent** across stop/restart, so
  `/var/lib/alpamon/alpamon.db` is preserved — no data loss.
- Idempotent with the existing tmpfiles / `ensureDirectories` paths.

### D. (recommended alongside) Harden `ensureDirectories()`

In `cmd/alpamon/command/register/service_linux.go`, inside the `HasSystemd()` branch
**only**, verify `DataDir()` exists after running `systemd-tmpfiles` and fall back to
a direct `mkdir` if not:

```go
func ensureDirectories() error {
	if utils.HasSystemd() {
		if output, err := exec.Command("systemd-tmpfiles", "--create", "alpamon.conf").CombinedOutput(); err != nil {
			return fmt.Errorf("tmpfiles creation failed: %w\n%s", err, string(output))
		}
		// Guard against systemd-tmpfiles exiting 0 without creating the dir
		// (e.g. unresolved drop-in). Otherwise the service crash-loops on 200/CHDIR.
		if _, err := os.Stat(utils.DataDir()); err != nil {
			return utils.EnsureDirectories()
		}
		return nil
	}
	return utils.EnsureDirectories()
}
```

The fallback stays inside the `HasSystemd()` branch so the non-systemd (container)
path is unchanged.

### Keep existing creation logic

The existing tmpfiles / `postinstall.sh` / `utils.EnsureDirectories()` paths must
**not** be removed — option A is an additional safety net, not a replacement (see
compatibility note below).

## Cross-platform impact

| Change | File | Build constraint | Container (no systemd) | Windows | macOS |
|---|---|---|---|---|---|
| A. StateDirectory | `configs/alpamon.service` | systemd unit (deb/rpm only) | installed but **unused** | **not installed / unused** | **not installed / unused** |
| D. ensureDirectories | `register/service_linux.go` | `_linux.go` → **compiled on GOOS=linux only** | isolated by `HasSystemd()` branch | not compiled | not compiled |

- **Container (no systemd)**: `ensureDirectories()` takes the `HasSystemd()==false`
  branch → `utils.EnsureDirectories()` (direct mkdir). `alpamon.service` is never
  read, so `StateDirectory` is inert.
- **Windows**: service is managed via Windows SCM (`service_windows.go`);
  `ensureDirectories()` → `utils.EnsureDirectories()`. `service_linux.go` and
  `alpamon.service` are not used. Paths are under `%ProgramData%\alpamon\…`.
- **macOS**: service is a launchd plist (`service_darwin.go`) whose
  `WorkingDirectory` is `/`, so the CHDIR issue does not exist;
  `ensureDirectories()` → `utils.EnsureDirectories()`.

→ The only behavior that changes is on systemd-based Linux hosts (the cloud EC2 case).

## Compatibility note (not a regression)

`StateDirectory` requires **systemd ≥ 235** (2017). On older systemd (RHEL/CentOS 7
= 219, Ubuntu 16.04 = 229) the directive is **ignored with an "unknown lvalue"
warning** — the unit still loads and starts, but the directory is not auto-created
there. Old systems remain covered by the existing tmpfiles / `ensureDirectories`
path (option D's fallback). Hence option A is purely additive: it helps modern
systemd and is harmless on old systemd.

## Acceptance criteria

- [ ] `configs/alpamon.service` declares `StateDirectory=alpamon` + `StateDirectoryMode=0750`.
- [ ] `ensureDirectories()` verifies `DataDir()` and falls back to `utils.EnsureDirectories()` inside the `HasSystemd()` branch.
- [ ] Existing tmpfiles / postinstall / `EnsureDirectories` paths are unchanged.
- [ ] On a clean systemd host with `/var/lib/alpamon` removed, `systemctl restart alpamon` reaches `active (running)` with no `200/CHDIR`, and `/var/lib/alpamon` is recreated `root:root 0750`.
- [ ] After `systemctl restart`, `/var/lib/alpamon/alpamon.db` is preserved (no data loss).
- [ ] `GOOS=linux|windows|darwin go build ./cmd/alpamon` all succeed.

## How to reproduce

```bash
# On a Linux + systemd host with alpamon installed:
sudo systemctl stop alpamon
sudo rm -rf /var/lib/alpamon
sudo systemctl start alpamon
systemctl status alpamon       # → failed, status=200/CHDIR
```

## Workaround (for already-broken instances)

```bash
sudo mkdir -p /var/lib/alpamon && sudo chmod 0750 /var/lib/alpamon
sudo systemctl reset-failed alpamon.service
sudo systemctl restart alpamon.service
systemctl status alpamon.service
```

## Related

- alpacon-server Cloud Phase 3-A (`install_script.sh`, SSM SendCommand) and
  Phase 3-D (`cloud_init_script.sh`, auto-provision). No server-side change is
  strictly required; optionally harden the script idempotency guard to also check
  `systemctl is-failed` so a "installed but failed" instance can self-recover
  (track as a separate alpacon-server issue).

### PR DESCRIPTION
**Pull request by geunwoonoh**
## Summary

Linux + systemd hosts installed through the Alpacon Cloud automated path (AWS SSM `AWS-RunShellScript` running `install_script.sh`, or Phase 3-D `cloud_init_script.sh`) install the deb/rpm and the systemd unit correctly, but the agent never reaches `active (running)`. `systemctl status alpamon` on a freshly provisioned EC2 shows the diagnostic:

```
(alpamon)[101306]: alpamon.service: Changing to the requested working directory failed: No such file or directory
(alpamon)[101306]: alpamon.service: Failed at step CHDIR spawning /usr/bin/alpamon: No such file or directory
systemd[1]: alpamon.service: Main process exited, code=exited, status=200/CHDIR
systemd[1]: alpamon.service: Failed with result 'exit-code'.
systemd[1]: alpamon.service: Scheduled restart job, restart counter is at 5.
systemd[1]: alpamon.service: Start request repeated too quickly.
systemd[1]: Failed to start alpamon.service - Secure Server Agent for Alpacon.
```

`status=200/CHDIR` is **not** a missing-binary symptom (`/usr/bin/alpamon` exists). It means systemd could not `chdir` into `WorkingDirectory=/var/lib/alpamon` before `ExecStart`. On the affected host, `ls -ld /var/lib/alpamon` confirms the directory does not exist.

The root cause is that the unit relied entirely on out-of-unit logic for `/var/lib/alpamon` to exist, and on the Cloud install path that logic can be skipped:

1. `apt-get install -y alpamon` runs `scripts/postinstall.sh`. Without `PLUGIN_ID`/`PLUGIN_KEY` env (the Cloud path uses flag-based `register` instead), `setup_alpamon()` returns non-zero, which short-circuits both `alpamon setup` and `start_systemd_service` and therefore never runs `systemd-tmpfiles --create`.
2. `alpamon register …` is then expected to create the directory via `cmd/alpamon/command/register/service_linux.go::ensureDirectories()` (which calls `systemd-tmpfiles --create alpamon.conf`).
3. If step 2's `systemd-tmpfiles` invocation does not actually create the directory (drop-in unresolved, old systemd silently ignoring options, etc.) but still exits `0`, `register` proceeds to `systemctl start` → immediate `200/CHDIR` crash loop.

Once it fails, the install script's idempotency guard (`systemctl is-active`) lets re-runs through, but `register.go` aborts at the "config file already exists" check **before** reaching `ensureDirectories()`, so the directory never gets created on retry either. The unit stays enabled and continues to crash-restart.

## Why this shipped without an obvious complaint loop

Two classes of installs were exercised before Cloud SSM, both of which silently mask the bug:

| Install path | `/var/lib/alpamon` created via | Hits the bug? |
|---|---|---|
| Manual install with `PLUGIN_ID`/`PLUGIN_KEY` env (legacy docs example) | `postinstall.sh` → `alpamon setup` → `systemd-tmpfiles --create` runs eagerly | no |
| Manual `alpamon register …` on a host that previously had alpamon installed | directory left over from a prior install | no |
| Containers (no systemd) | `utils.EnsureDirectories()` via the non-systemd branch of `ensureDirectories()` | no |
| Cloud SSM `install_script.sh` on a clean EC2 | `register` flag path → `systemd-tmpfiles` only. Silently no-ops on hosts where the drop-in is unresolved at register time. | **yes** |

The drop-in resolution timing on a fresh deb install in an SSM session was the missing piece in pre-release testing — none of the existing internal hosts had a clean `/var/lib/alpamon` absent state.

## Approach

Two complementary changes, both required.

**A. Declare `StateDirectory` on the unit.** systemd creates `StateDirectory=` **before** the `chdir` to `WorkingDirectory`. By telling systemd to own `/var/lib/alpamon`, we eliminate the CHDIR failure structurally on every service start, not just the register path:

```ini
StateDirectory=alpamon
StateDirectoryMode=0750
```

The directory is created `root:root 0750`, matching the existing `configs/tmpfile.conf` and `scripts/postinstall.sh::create_directories()` (no ownership or mode change for existing installs). Without `User=`/`DynamicUser=`, the directory is persistent across `stop`/`restart`, so the SQLite store at `/var/lib/alpamon/alpamon.db` is preserved.

**B. Harden `ensureDirectories()` in the register path.** `StateDirectory` requires systemd ≥ 235 (2017). On older systemd (RHEL/CentOS 7 = 219, Ubuntu 16.04 = 229) the directive is ignored with an "unknown lvalue" warning — the unit still loads, but the auto-create does not happen there. The register path also runs before `systemctl start`, so any fix that only kicks in at start time leaves the immediate `register` flow unguarded on those hosts. Inside the existing `HasSystemd()` branch, after `systemd-tmpfiles --create alpamon.conf`, we now `os.Stat(utils.DataDir())` and fall back to `utils.EnsureDirectories()` if the directory is still missing:

```go
if _, err := os.Stat(utils.DataDir()); err != nil {
    return utils.EnsureDirectories()
}
```

The fallback target `utils.EnsureDirectories()` is the same routine already used by `service_darwin.go`, `service_windows.go`, and the non-systemd branch — it creates all four alpamon directories (`/etc/alpamon` 0700, `/var/lib/alpamon` 0750, `/var/log/alpamon` 0750, `/run/alpamon` 0750) with `root:root` ownership.

### Rejected alternatives

- **`RuntimeDirectory=alpamon` for `/run/alpamon`**: default `RuntimeDirectoryPreserve=no` would delete the runtime directory on `stop`, destroying `auth.sock`/`logs.sock` between restarts. alpamon already `MkdirAll`s `/run/alpamon` at runtime, so the directive adds risk without benefit. **Not added.**
- **`LogsDirectory=alpamon` for `/var/log/alpamon`**: default mode `0755` mismatches `tmpfile.conf`'s `0750`, and alpamon writes to journald rather than that path. **Not added.**
- **`ConfigurationDirectory=alpamon` for `/etc/alpamon`**: missing `/etc/alpamon` surfaces as a config-load error in `pkg/config`, not as CHDIR, and is independently covered by `tmpfile.conf` and `postinstall.sh::create_directories()`. **Out of scope.**
- **Drop `WorkingDirectory=` from the unit**: `pkg/db/db.go` falls back to `filepath.Abs(dbFileName)` if `utils.DataDir()` is missing, which under systemd would put `alpamon.db` at `/` (systemd default CWD). The fallback exists for genuinely odd hosts and we do not want to make `/` the systemd default. **Rejected.**
- **Refactor `exec.Command` for unit-test injection**: the fallback target `utils.EnsureDirectories()` is already covered by `pkg/utils/systemd_test.go::TestEnsureDirectoriesWithRoot{,_Idempotent}`. The system-tmpfiles→stat→fallback composition is best verified by integration test on a real systemd host (see Test plan). **Not done in this PR.**

### `configs/alpamon.service`

Two-line addition inside `[Service]`, immediately after `WorkingDirectory=/var/lib/alpamon`:

```diff
 [Service]
 Type=simple
 ExecStart=/usr/bin/alpamon
 WorkingDirectory=/var/lib/alpamon
+StateDirectory=alpamon
+StateDirectoryMode=0750
 SELinuxContext=-unconfined_u:unconfined_r:unconfined_t:s0
 AppArmorProfile=-unconfined
```

No other unit directives are modified. `alpamon-restart.service` and `alpamon-restart.timer` are untouched (they only run `systemctl restart alpamon.service`, no CHDIR involved).

### `cmd/alpamon/command/register/service_linux.go`

`ensureDirectories()` gains a stat check + fallback inside the `HasSystemd()` branch. The non-systemd path (container without systemd PID 1) is byte-identical to before.

```diff
 func ensureDirectories() error {
 	if utils.HasSystemd() {
 		if output, err := exec.Command("systemd-tmpfiles", "--create", "alpamon.conf").CombinedOutput(); err != nil {
 			return fmt.Errorf("tmpfiles creation failed: %w\n%s", err, string(output))
 		}
+		// systemd-tmpfiles can exit 0 without creating the directory (e.g. on
+		// systemd <235 that ignores parts of the drop-in, or when the drop-in
+		// is unresolved). Without this fallback the service crash-loops on
+		// 200/CHDIR because WorkingDirectory does not exist.
+		if _, err := os.Stat(utils.DataDir()); err != nil {
+			return utils.EnsureDirectories()
+		}
 		return nil
 	}
 	return utils.EnsureDirectories()
 }
```

No new imports: `os` is already imported in this file.

### Files explicitly **not** touched

- `configs/tmpfile.conf`: canonical declarative directory list. Still the primary source for `systemd-tmpfiles`. Unchanged.
- `scripts/postinstall.sh`: `create_directories()` still covers non-systemd installs; `setup_alpamon()` env-gating is unchanged on purpose (this PR does not change the contract that env-less install requires explicit `register`).
- `pkg/utils/systemd.go`: `HasSystemd`, `EnsureDirectories`, `getAlpamonDirs` — fallback target only, no contract change.
- `pkg/db/db.go`: CWD fallback intentionally retained as a last-ditch safety net for non-systemd / unusual hosts. With this PR it is effectively unreachable on modern systemd, which is the intent.
- `cmd/alpamon/command/register/service_darwin.go`, `service_windows.go`: already short-circuit to `utils.EnsureDirectories()` and do not have a CHDIR surface (macOS launchd plist uses `WorkingDirectory=/`, Windows SCM has no CHDIR semantics).

## Behavior matrix

| Scenario | Before this PR | After this PR |
|---|---|---|
| Fresh systemd host, Cloud SSM install + `register` | ❌ `register` succeeds, service crash-loops with `status=200/CHDIR`, host stays disconnected | ✅ service reaches `active (running)`; `/var/lib/alpamon` is `root:root 0750` |
| `sudo rm -rf /var/lib/alpamon && sudo systemctl restart alpamon` on an installed host | ❌ same `200/CHDIR` loop (this is the reproduction recipe in the original issue) | ✅ systemd re-creates `/var/lib/alpamon` via `StateDirectory` before chdir |
| `systemctl restart alpamon` on a host where the directory already exists | ✅ works | ✅ unchanged: systemd leaves existing ownership/permissions intact per `systemd.exec(5)` |
| Old systemd (<235, e.g. RHEL 7 / Ubuntu 16.04) `alpamon register …` | ❌ same failure on the Cloud path | ✅ `StateDirectory` line is parsed and ignored with an "unknown lvalue" warning; `register`'s new `os.Stat` + `utils.EnsureDirectories()` fallback creates the directory |
| Container without systemd PID 1 (`HasSystemd()==false`) | ✅ `utils.EnsureDirectories()` direct path | ✅ unchanged (byte-identical branch) |
| macOS (launchd) | ✅ no CHDIR (plist `WorkingDirectory=/`) | ✅ unchanged |
| Windows (SCM) | ✅ no CHDIR semantics | ✅ unchanged |
| Restart preserves `/var/lib/alpamon/alpamon.db` | ✅ | ✅ unchanged: no `DynamicUser` ⇒ `StateDirectory` is persistent |

### Local

```bash
# Cross-compile sanity (acceptance criterion).
GOOS=linux   GOARCH=amd64 go build ./cmd/alpamon
GOOS=windows GOARCH=amd64 go build ./cmd/alpamon
GOOS=darwin  GOARCH=arm64 go build ./cmd/alpamon

# Full suite, sequential per CLAUDE.md.
go test ./... -p 1
```

Particular tests relevant to this change:

- `pkg/utils/systemd_test.go::TestEnsureDirectoriesWithRoot`
- `pkg/utils/systemd_test.go::TestEnsureDirectoriesWithRoot_Idempotent`
- `cmd/alpamon/command/register/...` (registration flow tests)

### Integration on a real systemd host

```bash
# 1. Install the new unit
sudo cp configs/alpamon.service /lib/systemd/system/
sudo systemctl daemon-reload

# 2. Reproduce the precondition the issue describes
sudo systemctl stop alpamon 2>/dev/null || true
sudo rm -rf /var/lib/alpamon
sudo systemctl reset-failed alpamon.service 2>/dev/null || true

# 3. Verify the fix
sudo systemctl restart alpamon
systemctl status alpamon                  # expect: active (running)
ls -ld /var/lib/alpamon                   # expect: drwxr-x--- root root
journalctl -u alpamon -n 50 --no-pager    # expect: no "200/CHDIR" lines

# 4. Data preservation across restart
stat -c '%i %Y' /var/lib/alpamon/alpamon.db
sudo systemctl restart alpamon
stat -c '%i %Y' /var/lib/alpamon/alpamon.db   # same inode, mtime unchanged
```

### Old-systemd compatibility spot check

On a RHEL 7 / Ubuntu 16.04 host (or any host with systemd < 235):

```bash
sudo systemctl daemon-reload 2>&1 | grep -i "unknown lvalue.*StateDirectory" || true
sudo systemctl restart alpamon
systemctl status alpamon                  # expect: active (running)
ls -ld /var/lib/alpamon
```

`daemon-reload` may emit an "unknown lvalue" warning for `StateDirectory`/`StateDirectoryMode`. The unit still loads. The register-path fallback ensures `/var/lib/alpamon` exists before `systemctl start`.

### Checklist

- [ ] `configs/alpamon.service` declares `StateDirectory=alpamon` and `StateDirectoryMode=0750`.
- [ ] `ensureDirectories()` calls `os.Stat(utils.DataDir())` and falls back to `utils.EnsureDirectories()` inside the `HasSystemd()` branch.
- [ ] Existing `tmpfile.conf` / `postinstall.sh` / `utils.EnsureDirectories` paths are byte-identical.
- [ ] On a clean systemd host with `/var/lib/alpamon` removed, `sudo systemctl restart alpamon` reaches `active (running)` with no `200/CHDIR`, and `/var/lib/alpamon` is recreated `root:root 0750`.
- [ ] After `sudo systemctl restart alpamon`, `/var/lib/alpamon/alpamon.db` is preserved (same inode).
- [ ] `GOOS=linux|windows|darwin go build ./cmd/alpamon` all succeed.
- [ ] `go test ./... -p 1` is green.
- [ ] On an old-systemd host, `daemon-reload` may warn about `StateDirectory`; the unit still loads and the register-time fallback creates the directory.

## Out of scope

- **`install_script.sh` idempotency guard on the alpacon-server side**. Today the guard only checks `systemctl is-active`; a host that is "installed but crash-looping" passes the guard but is not actually recoverable on retry because `register` then aborts at the "config file already exists" check. Hardening the guard to also check `systemctl is-failed` (and reset/restart in that case) is worth doing, but lives in `alpacon-server` `cloud/templates/cloud/install_script.sh` and is tracked separately. Importantly, alpamon's side of the fix in this PR is sufficient on its own — new installs no longer create the broken state.
- **Recovery on instances that already crash-looped before this fix lands**. Documented operator recovery (also in the issue): `sudo mkdir -p /var/lib/alpamon && sudo chmod 0750 /var/lib/alpamon && sudo systemctl reset-failed alpamon.service && sudo systemctl restart alpamon.service`. Reinstalling the deb after this PR is released is equivalent.
- **Adding `ConfigurationDirectory=alpamon` / `RuntimeDirectory=alpamon` / `LogsDirectory=alpamon`**. Rationale in "Rejected alternatives" above.

## References

- Issue: alpacax/alpamon#325 — "fix(systemd): alpamon.service fails with 200/CHDIR when /var/lib/alpamon does not exist".
- Affected components: `configs/alpamon.service`, `cmd/alpamon/command/register/service_linux.go`.
- systemd directives: `StateDirectory=`, `StateDirectoryMode=` — see `systemd.exec(5)`. Available since systemd 235 (2017).
- Related downstream: alpacon-server Cloud Phase 3-A (`install_script.sh`, SSM `SendCommand`) and Phase 3-D (`cloud_init_script.sh`, auto-provision). No server-side change is required for this fix to take effect; install scripts continue to invoke `alpamon register` unchanged.
- Field signature for triage: `journalctl -u alpamon` containing `Failed at step CHDIR spawning /usr/bin/alpamon: No such file or directory` and `Main process exited, code=exited, status=200/CHDIR`, with `ls -ld /var/lib/alpamon` returning `No such file or directory`.

### CODE REVIEW

File: cmd/alpamon/command/register/service_linux.go [hunk 1]
```diff
@@ -16,6 +16,13 @@ func ensureDirectories() error {
 		if output, err := exec.Command("systemd-tmpfiles", "--create", "alpamon.conf").CombinedOutput(); err != nil {
 			return fmt.Errorf("tmpfiles creation failed: %w\n%s", err, string(output))
 		}
+		// systemd-tmpfiles can exit 0 without creating the directory (e.g. on
+		// systemd <235 that ignores parts of the drop-in, or when the drop-in
+		// is unresolved). Without this fallback the service crash-loops on
+		// 200/CHDIR because WorkingDirectory does not exist.
+		if _, err := os.Stat(utils.DataDir()); err != nil {
+			return utils.EnsureDirectories()
+		}
```
[the remark this answers is not in the document]
Comment (geunwoonoh) [reply cmd/alpamon/command/register/service_linux.go]: 수정했습니다 (1e94c74).

- `os.IsNotExist(err)` 에만 `utils.EnsureDirectories()` 로 fallback 합니다. 이 케이스(systemd-tmpfiles 가 0 으로 종료하면서 디렉터리를 만들지 않는 경우)가 CHDIR fix 가 실제로 다뤄야 하는 시나리오입니다.
- 그 외 stat 에러(EACCES, EIO 등)는 `fmt.Errorf("stat %s: %w", DataDir, err)` 로 감싸서 그대로 반환 — register 시점에 즉시 드러나도록 했습니다.
- stat 이 성공했지만 디렉터리가 아닌 경우(잘못된 패키징으로 같은 경로에 파일이 있는 경우 등)에는 `"%s exists but is not a directory"` 로 명시적 에러를 반환합니다. unit 의 chdir 단계까지 가지 않고 register 가 먼저 멈춥니다.

**Review by jisung-02 — APPROVED** [review]
